### PR TITLE
feat(reload-middleware): Add reload-middleware

### PR DESCRIPTION
**Pull request by zdravko-georgiev**
#1042

### CODE REVIEW

File: packages/reload-middleware/README.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+|`https`| `object`| Configuration for starting the livereload server over https. The middleware supports automatically HTTP/2, when the UI5 server is started with the `--h2` option. In this case it uses by default `$HOME/.ui5/server/server.key` and `$HOME/.ui5/server/server.crt` |
+|`https.key`| `string`| Local filesystem path to the private key for https |
+|`https.cert`| `string`| Local filesystem path to the certificate for https |
+|`[connectOptions]`| `string \| string[] \| number \| boolean`| Configuration options of [connect-livereload](https://www.npmjs.com/package/connect-livereload#options) |
+
+## Configuration Examples
+
+### [Minimal configuration](#minimal-configuration)
+In order to use the middleware this is the minimal configuration that you need to provide in the `ui5.yaml` of your application. By default all files under the `webapp` folder with the following extensions `['html', 'js', 'ts', 'json','xml', 'properties', 'change', 'variant', 'ctrl_variant', 'ctrl_variant_change', 'ctrl_variant_management_change']` will be watched. The `livereload` server is started on port `35729`.
+
```
Comment (tobiasqueck) [inline packages/reload-middleware/README.md:22]: For completenes sake could you please add the snippet
```YAML
server:
  customMiddleware:
  - name: reload-middleware
     afterMiddleware: compression
```

Comment (zdravko-georgiev) [reply packages/reload-middleware/README.md:22]: Yes, will do

Comment (zdravko-georgiev) [reply packages/reload-middleware/README.md:22]: Done with https://github.com/SAP/open-ux-tools/pull/1704/commits/612b58a467185e15e56e86c559e3d69423bec118

File: packages/reload-middleware/README.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
+#  `@sap-ux/reload-middleware`
+
+The `@sap-ux/reload-middleware` is a [Custom UI5 Server Middleware](https://sap.github.io/ui5-tooling/pages/extensibility/CustomServerMiddleware) for reloading applications automatically during local development. Whenever a file relevant for the SAP Fiori applications is chaged the middleware will refresh the browser tab in which the application is running.
+
+It can be used either with the `ui5 serve` or the `fiori run` commands.
+
```
Comment (tobiasqueck) [inline packages/reload-middleware/README.md:6]: Could you please add a comment about it being used together with the `preview-middleware` and the required order of middlewares?
The same reference in the readme of the `preview-middleware` please

File: packages/reload-middleware/src/ui5/middleware.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import { resolve } from 'path';
+
+module.exports = async ({ options, middlewareUtil }: MiddlewareParameters<ReloaderConfig>): Promise<RequestHandler> => {
+    const logger = new ToolsLogger({
+        transports: [new UI5ToolingTransport({ moduleName: 'reload-middleware' })]
+    });
+    const rootPath = middlewareUtil.getProject().getRootPath();
+    const sourcePath = middlewareUtil.getProject().getSourcePath();
+    let httpsOpts: HttpsOptions = {
+        key: process.env.FIORI_TOOLS_SSL_KEY,
```
Comment (tobiasqueck) [inline packages/reload-middleware/src/ui5/middleware.ts:20]: Could you please mention the option of using these env variables in README.

File: packages/reload-middleware/src/ui5/middleware.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        cert: process.env.FIORI_TOOLS_SSL_CERT
+    };
+    if (options.configuration) {
```
Comment (tobiasqueck) [inline packages/reload-middleware/src/ui5/middleware.ts:23]: Interesting choice of implementation. I'd gone for setting the defaults on a configuration object if they are missing and then use only one implementation for the execution.
Nevertheless, your implementation is also ok.

**Review by tobiasqueck — CHANGES_REQUESTED** [review]
* changeset good
* sonar happy
* code is straight forward and easy to understand
* successfully tested it manually with both Fiori apps and adaptation projects
* Added a few comments for more information in the readme

File: packages/reload-middleware/README.md (hunk 2 above)
Comment (zdravko-georgiev) [reply packages/reload-middleware/README.md:6]: Yes, will do

File: packages/reload-middleware/src/ui5/middleware.ts (hunk 3 above)
Comment (zdravko-georgiev) [reply packages/reload-middleware/src/ui5/middleware.ts:20]: Yes, will do.

File: packages/reload-middleware/src/ui5/middleware.ts (hunk 4 above)
Comment (zdravko-georgiev) [reply packages/reload-middleware/src/ui5/middleware.ts:23]: thx

File: packages/reload-middleware/src/ui5/middleware.ts (hunk 3 above)
Comment (zdravko-georgiev) [reply packages/reload-middleware/src/ui5/middleware.ts:20]: Done with https://github.com/SAP/open-ux-tools/pull/1704/commits/f7b7191f4b98d7c912d89bc4bff471c4ee66c8f3

File: packages/reload-middleware/README.md (hunk 2 above)
Comment (zdravko-georgiev) [reply packages/reload-middleware/README.md:6]: Done with https://github.com/SAP/open-ux-tools/pull/1704/commits/f7b7191f4b98d7c912d89bc4bff471c4ee66c8f3

**Review by tobiasqueck — APPROVED** [review]
requested changes have been applied

**Review by tobiasqueck — APPROVED** [review]

**Review by Klaus-Keller — APPROVED** [review]
Thanks @zdravko-georgiev!
- code is well structured and easy to understand
- code coverage is good
- did not run local tests, just visual review

Approved from my side

**Review by tobiasqueck — APPROVED** [review]

**Pull request merged**